doc/api/cli.md

```diff
@@ -97,6 +97,43 @@ Automatically zero-fills all newly allocated [Buffer][] and [SlowBuffer][]
 instances.
 
 
+### `--preserve-symlinks`
+
+Instructs the module loader to preserve symbolic links when resolving and
+caching modules.
+
+By default, when Node.js loads a module from a path that is symbolically linked
+to a different on-disk location, Node.js will dereference the link and use the
+actual on-disk "real path" of the module as both an identifier and as a root
+path to locate other dependency modules. In most cases, this default behavior
+is acceptable. However, when using symbolically linked peer dependencies, as
+illustrated in the example below, the default behavior causes an exception to
+be thrown if `moduleA` attempts to require `moduleB` as a peer dependency:
+
+```text
+{appDir}
+ ├── app
+ │   ├── index.js
+ │   └── node_modules
+ │       ├── moduleA -> {appDir}/moduleA
+ │       └── moduleB
+ │           ├── index.js
+ │           └── package.json
+ └── moduleA
+     ├── index.js
+     └── package.json
+```
```

Comment:
In the pre-6.0 If `moduleA` requires `moduleB` to work, it will fail with the symbolic link consequently, regardless whether the "app/index.js" was the entry point or "moduleA" got imported from somewhere else.

With `--preserve-symlinks` the inclusion under "app/index.js" will start to work, the other one fails as it did before (and it should).

In some complex scenario this could lead to a serious head-scratching why this happens this way.